### PR TITLE
add explicit instructions to install page

### PR DESCRIPTION
**Pull request by bishabosha**
A script selects the default tab based on the OS (or 'Other' if JS is disabled). The accordion is also expanded by default if no JS is enabled.

Screenshot 2022-04-29 at 18 32 06

### CODE REVIEW

**Comment by bishabosha** [comment]
added some changes to use templates for reusable components

File: _layouts/download.html [hunk 1]
```diff
@@ -16,52 +16,100 @@ <h1>{{page.title}}</h1>
...
+{% assign currentScalaRelease = site.scalaversion %}
+{% for release in site.data.scala-releases %}
+  {% if release.category == 'current_version' %}
+    {% assign releaseMajor = release.version | truncate:1, '' %}
+    {% if releaseMajor == '3' %}
+      {% comment %}found latest Scala 3 release{% endcomment %}
+      {% assign currentScalaRelease = release.version %}
+    {% endif %}
+  {% endif %}
+{% endfor %}
```
Comment (julienrf) [inline _layouts/download.html:29]: Wouldn’t it be simpler to have a variable `site.scala3version` instead?

**Review by julienrf — CHANGES_REQUESTED** [review]
Thank you Jamie, I agree with the direction taken with this PR, but I have similar comments as with the docs.scala-lang PR regarding the stylesheets.

I also left a comment about the computation of the Scala 3 version.

**Comment by julienrf** [comment]
BTW, I had to re-generate the Gemfile.lock to be able to build the website. How does your lock file look like?

~~~ diff
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -1,95 +1,85 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    activesupport (5.2.4.3)
-      concurrent-ruby (~> 1.0, >= 1.0.2)
-      i18n (>= 0.7, < 2)
-      minitest (~> 5.1)
-      tzinfo (~> 1.1)
     addressable (2.8.0)
       public_suffix (>= 2.0.2, < 5.0)
     colorator (1.1.0)
-    colorize (0.8.1)
-    concurrent-ruby (1.1.7)
-    em-websocket (0.5.1)
+    concurrent-ruby (1.1.10)
+    em-websocket (0.5.3)
       eventmachine (>= 0.12.9)
-      http_parser.rb (~> 0.6.0)
-    ethon (0.11.0)
-      ffi (>= 1.3.0)
+      http_parser.rb (~> 0)
+    ethon (0.15.0)
+      ffi (>= 1.15.0)
     eventmachine (1.2.7)
-    ffi (1.13.1)
+    ffi (1.15.5)
     forwardable-extended (2.6.0)
-    html-proofer (3.10.2)
-      activesupport (>= 4.2, < 6.0)
+    html-proofer (3.19.3)
       addressable (~> 2.3)
-      colorize (~> 0.8)
-      mercenary (~> 0.3.2)
-      nokogiri (~> 1.9)
+      mercenary (~> 0.3)
+      nokogiri (~> 1.12)
       parallel (~> 1.3)
+      rainbow (~> 3.0)
       typhoeus (~> 1.3)
       yell (~> 2.0)
-    http_parser.rb (0.6.0)
-    i18n (0.9.5)
+    http_parser.rb (0.8.0)
+    i18n (1.10.0)
       concurrent-ruby (~> 1.0)
-    jekyll (3.9.0)
+    jekyll (4.2.2)
       addressable (~> 2.4)
       colorator (~> 1.0)
       em-websocket (~> 0.5)
-      i18n (~> 0.7)
-      jekyll-sass-converter (~> 1.0)
+      i18n (~> 1.0)
+      jekyll-sass-converter (~> 2.0)
       jekyll-watch (~> 2.0)
-      kramdown (>= 1.17, < 3)
+      kramdown (~> 2.3)
+      kramdown-parser-gfm (~> 1.0)
       liquid (~> 4.0)
-      mercenary (~> 0.3.3)
+      mercenary (~> 0.4.0)
       pathutil (~> 0.9)
-      rouge (>= 1.7, < 4)
+      rouge (~> 3.0)
       safe_yaml (~> 1.0)
-    jekyll-redirect-from (0.14.0)
-      jekyll (~> 3.3)
-    jekyll-sass-converter (1.5.2)
-      sass (~> 3.4)
+      terminal-table (~> 2.0)
+    jekyll-redirect-from (0.16.0)
+      jekyll (>= 3.3, < 5.0)
+    jekyll-sass-converter (2.2.0)
+      sassc (> 2.0.1, < 3.0)
     jekyll-watch (2.2.1)
       listen (~> 3.0)
-    kramdown (2.3.1)
+    kramdown (2.4.0)
       rexml
     kramdown-parser-gfm (1.1.0)
       kramdown (~> 2.0)
     liquid (4.0.3)
-    listen (3.2.1)
+    listen (3.7.1)
       rb-fsevent (~> 0.10, >= 0.10.3)
       rb-inotify (~> 0.9, >= 0.9.10)
-    mercenary (0.3.6)
-    mini_portile2 (2.8.0)
-    minitest (5.14.1)
-    nokogiri (1.13.4)
-      mini_portile2 (~> 2.8.0)
+    mercenary (0.4.0)
+    nokogiri (1.13.4-x86_64-linux)
       racc (~> 1.4)
-    parallel (1.12.1)
+    parallel (1.22.1)
     pathutil (0.16.2)
       forwardable-extended (~> 2.6)
-    public_suffix (4.0.6)
+    public_suffix (4.0.7)
     racc (1.6.0)
-    rake (12.3.3)
-    rb-fsevent (0.10.4)
+    rainbow (3.1.1)
+    rake (13.0.6)
+    rb-fsevent (0.11.1)
     rb-inotify (0.10.1)
       ffi (~> 1.0)
     rexml (3.2.5)
-    rouge (3.22.0)
+    rouge (3.28.0)
     safe_yaml (1.0.5)
-    sass (3.7.4)
-      sass-listen (~> 4.0.0)
-    sass-listen (4.0.0)
-      rb-fsevent (~> 0.9, >= 0.9.4)
-      rb-inotify (~> 0.9, >= 0.9.7)
-    thread_safe (0.3.6)
-    typhoeus (1.3.1)
+    sassc (2.4.0)
+      ffi (~> 1.9)
+    terminal-table (2.0.0)
+      unicode-display_width (~> 1.1, >= 1.1.1)
+    typhoeus (1.4.0)
       ethon (>= 0.9.0)
-    tzinfo (1.2.7)
-      thread_safe (~> 0.1)
-    yell (2.0.7)
+    unicode-display_width (1.8.0)
+    yell (2.2.2)
 
 PLATFORMS
-  ruby
+  x86_64-linux
 
 DEPENDENCIES
   html-proofer
@@ -100,4 +90,4 @@ DEPENDENCIES
   rake
 
 BUNDLED WITH
-   2.1.4
+   2.3.10
~~~

File: _layouts/download.html (hunk 1 above)
Comment (bishabosha) [reply _layouts/download.html:29]: that would also be sensible, I didn't originally just as would add another step for the release process

**Comment by bishabosha** [comment]
each new js click listener is now added dynamically when the document loads (rather than using "onclick").

**Comment by SethTisue** [comment]
> BTW, I had to re-generate the Gemfile.lock to be able to build the website. How does your lock file look like?

Pulling in #1377 here might fix that?

**Comment by julienrf** [comment]
@SethTisue Probably, yes, sorry I missed that PR!

**Comment by bishabosha** [comment]
Here are the new styling changes: (tabs are reused from the blog page)

Screenshot 2022-05-06 at 19 31 51

File: _layouts/download.html (hunk 1 above)
Comment (julienrf) [reply _layouts/download.html:29]: OK.

**Review by julienrf — CHANGES_REQUESTED** [review]
The result looks great!

I noticed two issues still:

- the mouse cursor, when we hover on a tab, is not a pointer. I think it should be,
- the copy button does not copy the command to my clipboard (linux + firefox)

**Review by julienrf — APPROVED** [review]

**Comment by bishabosha** [comment]
@julienrf I have added an extra commit that unifies the code highlighting for the `text-step` (replacing `text-area-code`), so you should check that the individual release pages are satisfactory

**Pull request merged**